junifer/storage/hdf5.py

```diff
@@ -0,0 +1,849 @@
+"""Provide concrete implementation for feature storage via HDF5."""
+
+# Authors: Synchon Mandal <s.mandal@fz-juelich.de>
+#          Federico Raimondo <f.raimondo@fz-juelich.de>
+# License: AGPL
+
+
+from pathlib import Path
+from typing import Any, Dict, Iterable, List, Optional, Union
+
+import numpy as np
+import pandas as pd
+from h5io import read_hdf5, write_hdf5
+from tqdm import tqdm
+
+from ..api.decorators import register_storage
+from ..utils import logger, raise_error
+from .base import BaseFeatureStorage
+from .utils import element_to_prefix
+
+
+@register_storage
+class HDF5FeatureStorage(BaseFeatureStorage):
+    """Concrete implementation for feature storage via HDF5.
+
+    Parameters
+    ----------
+    uri : str or pathlib.Path
+        The path to the file to be used.
+    single_output : bool, optional
+        If False, will create one HDF5 file per element. The name
+        of the file will be prefixed with the respective element.
+        If True, will create only one HDF5 file as specified in the
+        ``uri`` and store all the elements in the same file. Concurrent
+        writes should be handled with care (default True).
+    overwrite : bool or "update", optional
+        Whether to overwrite existing file. If True, will overwrite and
+        if "update", will update existing entry or append (default "update").
+    compression : {0-9}, optional
+        Level of gzip compression: 0 (lowest) to 9 (highest) (default 7).
+    **kwargs : dict
```

Comment:
This still needs to be addressed.

Reply:
We have `**kwargs` in the constructor signature for smooth subclassing and just makes it transparent to the reader that it has it. I don't have a preference here, but I have always found it in the wild and is natural to me at this point. If you have a strong reason, I don't mind.

Reply:
If new parameters arise, then we can change the classes. We are mainly focused on user-oriented documentation, so I want to avoid having things that we don't document / are difficult to follow from the end-user point of view.

Reply:
Do you mean to remove it from the signature as well as docstring or just the docstring?

Reply:
Addressed in `18643864`.